master/internal/proxy/proxy.go

```diff
@@ -74,19 +87,23 @@ func (p *Proxy) Receive(ctx *actor.Context) error {
 	return nil
 }
 
-func (p *Proxy) getTarget(serviceName string) (*url.URL, bool) {
-	p.lock.RLock()
-	defer p.lock.RUnlock()
-	target, ok := p.services[serviceName]
-	return target, ok
+func (p *Proxy) getTargetURL(serviceName string) (*url.URL, bool) {
+	p.lock.Lock()
```

Comment:
question: I know its not in scope for this PR, but why do we even need this lock?

Reply:
`p.services` is a shared map and therefore, unregistering a service while reading from the map could result in concurrent access to the map.